src/IO/Exporter/PointwiseInterpolator.cpp

```diff
@@ -582,6 +599,136 @@ void interpolate_to_points(
   }
 }
 
+template <size_t Dim, typename Frame>
+PointwiseInterpolator<Dim, Frame>::PointwiseInterpolator(
+    const std::variant<std::vector<std::string>, std::string>&
+        volume_files_or_glob,
+    const std::string& subfile_name, const ObservationVariant& observation,
+    const std::vector<std::string>& tensor_components) {
+  auto bindings =
+      parse_volume_files<Dim>(volume_files_or_glob, subfile_name, observation);
+  const auto& filenames = std::get<0>(bindings);
+  obs_id_ = std::get<1>(bindings);
+  time_ = std::get<2>(bindings);
+  domain_ = std::move(std::get<3>(bindings));
+  functions_of_time_ = std::move(std::get<4>(bindings));
+
+  // Load tensor data into memory
+  element_ids_.reserve(filenames.size());
+  meshes_.reserve(filenames.size());
+  tensor_data_.reserve(filenames.size());
+  for (const auto& filename : filenames) {
+    const h5::H5File<h5::AccessType::ReadOnly> h5file(filename);
+    const auto& volfile = h5file.get<h5::VolumeData>(subfile_name);
+    const auto [element_ids, meshes] = load_grids<Dim>(volfile, obs_id_);
+    element_ids_.push_back(std::move(element_ids));
+    meshes_.push_back(std::move(meshes));
+    auto tensor_data = load_tensor_data(volfile, obs_id_, tensor_components);
+    tensor_data_.push_back(std::move(tensor_data));
+    h5file.close();
+  }
+}
+
+template <size_t Dim, typename Frame>
+void PointwiseInterpolator<Dim, Frame>::interpolate_to_points(
+    const gsl::not_null<std::vector<DataVector>*> result,
+    const tnsr::I<DataVector, Dim, Frame>& target_points,
+    const bool extrapolate_into_excisions, const bool error_on_missing_points,
+    const std::optional<size_t> num_threads) const {
+  ASSERT(not tensor_data_.empty(),
+         "PointwiseInterpolator has not been initialized with tensor data.");
```

Comment:
As the class isn't serializable, is there a reason to allow the invalid default-constructed state?

Reply:
I needed it for default construction in a class that is serializable and holds an interpolator as member variable (but doesn't serialize it).